src/Request/TextChatGenerationFromScratch.php

```diff
@@ -77,7 +77,8 @@ public function prepareResponse(array $response): ResponseInterface
         $texts = [];
         foreach ($response['choices'] as $item) {
             $texts[] = new Chat($item['message']['role'], $item['message']['content'], $item['index'], $item['finish_reason']);
-        }
-        return new Texts($texts, $response['created'], $response['model'], $response['id']);
+        };
+
+        return new Texts($texts, $response['created'], $response['model'], $response['id'], (new PriceCalculation)->calculatePrice($response['model'], $response['usage']));
```

Comment:
`(new PriceCalculation)->calculatePrice($response['model'], $response['usage'])` Przypisz proszę wynik do zmiennej, a następnie dodaj tę zmienną do obiektu. To zwiększy czytelność kodu.